src/app/store/trips/reducer.ts

```diff
@@ -0,0 +1,105 @@
+import { isDevMode } from '@angular/core';
+import { createReducer, MetaReducer, on } from '@ngrx/store';
+import { Trip } from '../../models/trips';
+import {
+  createNewTrip,
+  createNewTripComplete,
+  createNewTripError,
+  deleteTrip,
+  deleteTripComplete,
+  deleteTripError,
+  getAllTrips,
+  getAllTripsComplete,
+  getAllTripsError,
+  setSelectedTripId,
+  updateTrip,
+  updateTripComplete,
+  updateTripError,
+} from './actions';
+
+export const tripsFeatureKey = 'Trips';
+
+export interface TripState {
+  trips: Trip[];
+  isLoading: boolean;
+  selectedTripId?: string;
+}
+
+const initialState: TripState = {
+  trips: [],
+  isLoading: false,
+};
+
+export const tripsReducers = createReducer(
+  initialState,
+  on(getAllTrips, state => ({
+    ...state,
+    isLoading: true,
+  })),
+  on(getAllTripsComplete, (state, { trips }) => ({
+    ...state,
+    isLoading: false,
+    trips,
+  })),
+  on(getAllTripsError, (state, { error }) => {
+    alert(`Error getting trips. Error message: ${error.message}`);
```

Comment:
I personally dislike this here. Alert is a side-effect. Side effects belong in effects.

Reply:
Cool, will move it to effect 👍🏻